bundles/org.openhab.binding.venstarthermostat/src/main/java/org/openhab/binding/venstarthermostat/internal/handler/VenstarThermostatHandler.java

```diff
@@ -415,6 +474,51 @@ private void updateData() {
             updateIfChanged(CHANNEL_EXTERNAL_TEMPERATURE, getOutdoorTemperature());
             updateIfChanged(CHANNEL_HUMIDITY, getHumidity());
 
+            response = getData("/query/runtimes");
+            if (!isFutureValid(localUpdatesTask)) {
+                return;
+            }
+
+            runtimeData = Objects.requireNonNull(gson.fromJson(response, VenstarRuntimeData.class));
+            List<VenstarRuntime> runtimes = runtimeData.getRuntimes();
+            String[] timestamps = { CHANNEL_TIMESTAMP_RUNTIME_DAY0, CHANNEL_TIMESTAMP_RUNTIME_DAY1,
```

Comment:
I agree it's getting messy. I didn't like it when I programmed it, but I'm not sure how to resolve it (your CHANNEL_NAME+i won't work, will it?). I was actually wondering if it would be cleaner just to provide the runtime string, and let users do whatever they want with it (with some explanation of the String content in the documentation). I suspect anyone trying to do anything with this data will be quite "hardcore" anyway.

Reply:
Thanks!! Sounds good....I will update it. I guess I had somehow forgotten the variable values are strings.

Reply:
What do you think about just providing the runtime string as it comes directly from the thermostat, as a simplification? I suppose as I've already done a lot of code around this, it's not worth reversing :-)

Reply:
> What do you think about just providing the runtime string as it comes directly from the thermostat,

Not sure what you mean?

Reply:
I was just thinking it would be simpler to provide the raw json response from '.../query/runtimes' and let people do whatever they want with it with a JSON / Regex transformation operation. That would avoid all the extra channels and code we currently have (I don't like the fact the binding currently has 56 channels and 2 classes for this - it's messy :-O). A json probably wouldn't be too difficult to handle for people that are likely to use the data (it's not really core thermostat functionality). The thermostat also gives alarms in a similar format, which would again require multiple channels and classes unless a JSON response is used.

Reply:
No, returning raw JSON is not a great choice here.  I am fine with the extra channels and classes, you can mark them as advanced, and they won't show up by default in the UI, so it should not intimidate  users.  Returning in JSON will ensure very few people would ever use this.  Again, if you dynamically create the channel names as we discussed a few times, then its just the thing XML which is long, but there are definitely bindings with more in them, so not out of the ordinary .

Reply:
@digitaldan OK. I did the updates you suggested anyway. They are in my latest commit/push in this pull request.